src/protocols/secure_channel/SessionParameters.h

```diff
@@ -0,0 +1,91 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines a Session Parameters that are passed during session estalishment.
+ *
+ */
+
+#pragma once
+
+#include <lib/core/TLV.h>
+#include <messaging/ReliableMessageProtocolConfig.h>
+
+namespace chip {
+
+// TODO We should get part of this from constexpr that is in ReliableMessageProtocolConfig.h
+
+class SessionParameters
+{
+public:
+    SessionParameters(ReliableMessageProtocolConfig mrpConfig = GetDefaultMRPConfig()) : mMRPConfig(mrpConfig) {}
+
+    // This estimated TLV size calc is here instead of messaging/ReliableMessageProtocolConfig.h
+    // because we would need to add `include <lib/core/TLV.h>`. While we could make it all work
+    // from a build standpoint, if any new MRP config gets added accessors will still need to be
+    // added here so having this calc done here isn't problematic.
+    static constexpr size_t kMrpConfigEstimatedTLVSize =
+        TLV::EstimateStructOverhead(sizeof(uint32_t), sizeof(uint32_t), sizeof(uint16_t));
+
+    static constexpr size_t kEstimatedTLVSize = kMrpConfigEstimatedTLVSize +
```

Comment:
would it be useful to see what we estimate here (in case we eventually add IDLE/ACTIVE parameters as well?

like data_model_revision, im_revision, specification_version and command_count?